curriculum/challenges/english/25-front-end-development/lecture-working-with-data-fetching-and-memoization-in-react/67d1a99d10fd509c88faf3bf.md

```diff
@@ -2,13 +2,300 @@
 id: 67d1a99d10fd509c88faf3bf
 title: How Does Data Fetching Work in React?
 challengeType: 11
-videoId: nVAaxZ34khk
+videoId: _gwWcrZ1WJo
 dashedName: how-does-data-fetching-work-in-react
 ---
 
 # --description--
 
-Watch the video lecture and answer the questions below.
+Watch the video or read the transcript and answer the questions below.
+
+# --transcript--
+
+How does data fetching work in React?
+
+React apps often rely on external APIs and databases for dynamic content. To access the data from those APIs and databases, you need to use some data fetching techniques.
+
+Let's take a look at how data fetching works in React and the different options available to you for fetching data.
+
+React is not opinionated about how you fetch data, this means on a basic level, you can use the built-in Fetch API, which all modern browsers support.
+
+You can also use Axios and SWR. Axios is promise-based HTTP request library built on top of the XMLHttpRequest object, and SWR is a React hook for data fetching created by the Vercel team.
+
+Let's start with an example. You first need to import the `useState` and `useEffect` hooks:
+
+```js
+import { useState, useEffect } from "react";
+```
+
+Then you will need to create three state variables called `loading`, `data`, and `error`:
+
+```js
+const [data, setData] = useState(null);
+const [loading, setLoading] = useState(true);
+const [error, setError] = useState(null);
+```
+
+The `loading` variable will track whether the data is still being fetched. The `data` variable represents the data itself, and the `error` variable will capture any errors that might occur during the data fetching process.
+
+Since data fetching is a side effect, it's best to use the fetch API inside of a `useEffect` hook.
```

Comment:
not sure if fetch API needs to be in backticks

Reply:
Also, it looks like there are inconsistent uses for fetch API.

Sometimes it is written fetch API.
Sometimes it is Fetch API.

So we have to pick one version and stick with that

Reply:
Fixed, they should all be "Fetch API" now. I usually just check MDN or official docs if I'm not sure - MDN uses "Fetch API"